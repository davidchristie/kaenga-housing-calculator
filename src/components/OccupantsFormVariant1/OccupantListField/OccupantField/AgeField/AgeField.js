import React, { Component, PropTypes } from 'react'
import {
  ControlLabel,
  FormControl,
  FormGroup
} from 'react-bootstrap'

class AgeField extends Component {
  render () {
    const { input: { onChange, value } } = this.props
    return (
      <FormGroup>
        <ControlLabel>How old is the occupant?</ControlLabel>
        <FormControl
          componentClass='select'
          onChange={onChange}
          value={value}
        >
          <option value='adult'>Adult</option>
          <option value='teenager'>Teenager</option>
          <option value='child'>Child</option>
        </FormControl>
      </FormGroup>
    )
  }
}

AgeField.defaultProps = {
  input: {}
}

AgeField.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.string
  }).isRequired
}

export default AgeField
