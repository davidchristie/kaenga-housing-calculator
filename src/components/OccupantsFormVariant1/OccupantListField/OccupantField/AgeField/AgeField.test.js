import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import AgeField from './AgeField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<AgeField />)).toMatchSnapshot()
})
