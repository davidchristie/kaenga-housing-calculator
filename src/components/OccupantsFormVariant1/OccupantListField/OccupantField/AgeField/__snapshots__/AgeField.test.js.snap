exports[`test matches snapshot 1`] = `
<FormGroup
  bsClass="form-group">
  <ControlLabel
    bsClass="control-label"
    srOnly={false}>
    How old is the occupant?
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="select"
    onChange={undefined}
    value={undefined}>
    <option
      value="adult">
      Adult
    </option>
    <option
      value="teenager">
      Teenager
    </option>
    <option
      value="child">
      Child
    </option>
  </FormControl>
</FormGroup>
`;
